src/content/tools/widget-previewer.md

```diff
@@ -0,0 +1,150 @@
+
+```dart
+import 'package:flutter/widget_previews.dart';
+import 'package:flutter/material.dart'; // For Material widgets
+
+@Preview(name: 'My Sample Text')
+Widget mySampleText() {
+  return const Text('Hello, World!');
+}
+```
+
+![Sample widget in Flutter Widget Previewer](/assets/images/docs/tools/widget-previewer/widget-previewer.png "Example widget"){:width="50%"}
+
+Each preview instance provides various controls for
+interacting with the previewed widget. From left to right:
+
+*   **Zoom in:** Magnifies the widget in the preview.
+
+*   **Zoom out:** Reduces the magnification of the widget in
+    the preview.
+
+*   **Reset zoom:** Returns the widget preview to its
+    default zoom level.
+
+*   **Toggle between light and dark mode:** Switches the
+    preview's theme between a light and dark color scheme.
+
+*   **Perform a hot restart for the individual preview:**
+    Restarts only the specific widget preview,
+    allowing changes to be applied quickly without
+    restarting the entire application.
+
+For the case where global state has been modified
+(for example, a static initializer has been changed), the
+entire widget previewer can be told to hot restart using the
+button at the top right of the environment.
+
+## Customize a preview
+
+The [`@Preview`][] annotation has several parameters you can
+use to customize the preview:
+
+*   **`name`**: A descriptive name for the preview.
+
+*   **`size`**: Artificial size constraints using a Size
+    object.
+  
+*   **`textScaleFactor`**: A custom font scale.
+
+*   **`wrapper`**: A function that wraps your previewed
+    widget in a specific widget tree (for example, to inject
+    application state into the widget tree via a Provider).
+  
+*   **`theme`**: A function to provide Material and
+    Cupertino theming data.
+  
+*   **`brightness`**: The initial theme brightness.
+
+*   **`localizations`**: A function to apply a localization
+    configuration.
+
+[`@Preview`]: {{site.repo.flutter}}/blob/main/packages/flutter/lib/src/widget_previews/widget_previews.dart
+
+## Restrictions and Limitations
+
+The Flutter Widget Previewer has certain restrictions you
+should be aware of:
+
+*   **Public Constants**: All arguments provided to the
+    `@Preview` annotation must be public and constant.
+    This is required for the previewer's code generation
+    implementation to work correctly. The requirement for
+    public variable names will be relaxed in future
+    releases, but function arguments must always have
+    public names.  
+
+*   **Unsupported APIs**: Native plugins and any APIs from
+    the `dart:io` library are not supported. This is because
+    the widget previewer is built with
+    Flutter Web, which doesn't have access to the underlying
+    native platform APIs. While web plugins may work when
+    using Chrome, there is no guarantee that they will work
+    within IDEs when support is added in a future release.  
```

Comment:
I think we don't need to discuss a potential future release here. We can instead make the note more generic and avoid necessary adjustments if such a release is made. 

```suggestion
    using Chrome, there is no guarantee that they will work
    within other environments, such as when embedded in IDEs.
```